File: 509-Fibonacci Number-斐波那契数/509-Fibonacci Number-斐波那契数.java
class Solution {
    public int fib(int n) {
        if(n == 0) return 0;
        int x = 0;
        int y = 1;
        int temp;
        for(int i = 0; i < n - 1; i++){
            temp = x + y;
            x = y;
            y = temp;
        }
        return y;
    }
}

File: 509-Fibonacci Number-斐波那契数/509-Fibonacci Number-斐波那契数.js
/**
 * @param {number} n
 * @return {number}
 */
 var fib = function(n) {
    if(n === 0) return 0
    let dp = [0, 1]
    let temp
    for(let i = 0; i < n - 1; i++){
        temp = dp[0] + dp[1]
        dp[0] = dp[1]
        dp[1] = temp
    } 
    return dp[1]
};